packages/effect/dtslint/Array.tst.ts

```diff
@@ -355,10 +351,8 @@ describe("Array", () => {
       })
     )).type.toBe<Array<string | number>>()
 
-    // @ts-expect-error: wrong `_item` type
-    Array.filter(numbersOrStrings, (_item: string) => true)
-    // @ts-expect-error: wrong `_item` type
-    pipe(numbersOrStrings, Array.filter((_item: string) => true))
+    expect(Array.filter).type.not.toBeCallableWith(numbersOrStrings, (_item: string) => true)
+    expect(Array.filter<string | number, string | number>).type.not.toBeCallableWith((_item: string) => true)
```

Comment:
Type arguments must be passed to `Array.filter` to reproduce the same behaviour. Perhaps there are better solutions?

The following does not work, because the error originates from `Array.filter`. It is reported as a problem.

```ts
expect(pipe).type.not.toBeCallableWith(numbersOrStrings, Array.filter((_item: string) => true))
```

The assertion can be moved to the argument list of `pipe()`. This will error as well, because `pipe()` cannot that `expect()` as an argument:

```ts
pipe(numbersOrStrings, expect(Array.filter).type.not.toBeCallableWith((_item: string) => true))
```

Reply:
By the way, I think `// @ts-` comments must be removed in the virtual file. They will hide errors in erased expressions and matchers will not work.

Cases like this one could use the ignore comment. And then the first solution above works as expected.

Reply:
A helper function could be an elegant solution. Instead of making noise here, I opened https://github.com/tstyche/tstyche/issues/458